### DISCUSSION TITLE
🚨 Xcode 14.3 and possible breaking changes 🚨

### DISCUSSION THREAD

**Discussion by mikhail-tokarev** [codemagic-ci-cd/codemagic-docs#1545] [Announcements]
As you may already know, Apple released Xcode 14.3 on March 31. This version requires a Mac running macOS Ventura 13.0 or later, which may cause breaking changes when switching from a macOS Monterey 12.x build machine. Please ensure your Xcode projects are updated locally before updating CI/CD configuration on Codemagic.

Additionally, Xcode 14.3 is only available on Mac mini M1 machines and is not available on Intel-based Mac Pro or older machines. The highest available Xcode version on Intel-based machines is Xcode 14.2. Please review your build configuration to avoid build failures. 

**On April 19, we will change the default behavior for builds configured to use Xcode `latest` or` edge` version and run them using the Xcode 14.3 Release version.** If you already build with Xcode 14.3, no changes are required from your side.

See the [build machine specs](https://docs.codemagic.io/specs/versions-macos/) for information about the preinstalled software and available runtimes.

If you run into any unexpected behavior or any other problems when upgrading your workflow to use Mac mini M1 build machine, please let us know in this thread or by reaching out to support via the in-app Chat widget.

**Comment by lebasp** [comment]
The Unity builds weren't working on the M1 instances the last time we tried, all the projects we tested didn't compile, we changed to macbook pro and it works without changing anything else.

The support asked us to change to mac pro instances.

Will there be macbook pro m1 to use on Unity projects?

**Reply by mikhail-tokarev** [reply]
Thank you for your question.

There is a Unity-side limitation on building Unity projects on M1 machines with full ARM support. We still recommend using Intel-based Mac Pro and Xcode 14.2 until Unity will fix the issue.

**Reply by lebasp** [reply]
Is it about the mac mini itself or the virtualization of the machines?
 
I asked about it because we are using physical M1 Macbook Pro 2020 (Ventura) and Unity compiles it successfully locally.

**Reply by mikhail-tokarev** [reply]
it's about nested virtualisation on M1

**Reply by lebasp** [reply]
O I see, did you guys contacted unity about it already?

**Reply by martinjeret** [reply]
@lebasp we did contact Unity, sorry for the radio silence. We are not able to share the ticket since it's private :(

**Comment by danallen88** [comment]
> Additionally, Xcode 14.3 is only available on Mac mini M1 machines and is not available on Intel-based Mac Pro or older machines.

Is this a limitation being imposed by Codemagic? I am running Xcode 14.3 on an Intel based MacBook Pro with macOS 13.2.1 without issues currently.

**Reply by martinjeret** [reply]
Unfortunately the Mac Pros we have are not compatible with macOS Ventura and therefore we just simply cannot install it and macOS Ventura is required for Xcode 14.3. This unfortunately is a limitation introduced by Apple. Thankfully we have the M1 Mac machines available to support latest macOS versions.

**Comment by azun-k** [comment]
Am I STILL able to specify Xcode 14.2 for my builds on April 19 onwards?

**Reply by martinjeret** [reply]
Yes

**Reply by azun-khang** [reply]
Hi @martinjeret , for now, is there a way to try with Xcode 14.3?
I followed the instructions [here](https://docs.codemagic.io/yaml-basic-configuration/yaml-getting-started/), 
with a bit customization: 
```
definitions: 
  env_versions: &env_versions
    xcode: 14.3 
    cocoapods: default
```
instead of `xcode: latest`, i used `xcode: 14.3`. 
CodeMagic build showed the error:
`The mac_pro instance type does not support the selected Xcode version. Please consider using mac_mini_m1 instead.`
<img width="633" alt="mac_mini_m1" src="https://user-images.githubusercontent.com/103913863/232362042-e28b4f8b-feef-4073-baa2-ddce32681d90.png">

Again, how can i verify CodeMagic builds with Xcode 14.3 before April 19?
Thanks

**Reply by martinjeret** [reply]
you need to change your `instance type` to `mac_mini_m1 `

https://docs.codemagic.io/yaml-basic-configuration/yaml-getting-started/#instance-type

**Reply by azun-khang** [reply]
Thanks @martinjeret .

**Comment by gerken-tss** [comment]
There is an issue that arises when using Xcode 14.3 and pods that have a `IPHONEOS_DEPLOYMENT_TARGET` less than 11.0. To circumvent this problem, you can use [this solution I have posted on StackOverflow](https://stackoverflow.com/a/76045501/2102328).

It basically stets the `IPHONEOS_DEPLOYMENT_TARGET` of every pod to the version you specify under `wanted_project_target` if the original version of this pod is lower (in the `post_install` hook).

If I'm not mistaken, this issue should happen in the Codemagic CI pipeline as well and this will be the case for many projects when `latest` and `edge` point to Xcode 14.3 (which is tomorrow).

**Reply by icarusdust** [reply]
@gerken-tss, looks like Flutter released a fix for this with 3.7.11 : https://github.com/flutter/flutter/issues/124340

**Reply by gerken-tss** [reply]
@icarusdust awesome! Thank you for noticing

**Comment by helinanever** [comment]
Update: `latest` and `edge` in Xcode version settings now point to 14.3.

**Comment by montreux** [comment]
Could we please update the build machine specs to use cocoapods 1.12.1, as it fixes a conflict preventing IPA builds when using cocoapods 1.12.0 and Xcode 14.3. Currently projects using cocoapods will fail with the default build settings: `'Build ipa for distribution' exited with status code 65` 

I mentioned in other places that a workaround for now is to replace `cocoapods: default` with `cocoapods: 1.12.1` in codemagic.yaml.
